TheWarWithin/Priorities/MageFrost.simc

```diff
@@ -46,64 +65,72 @@ actions.cds+=/lights_judgment
+actions.cleave_ss+=/freeze,if=freezable&debuff.frozen.down&(!talent.glacial_spike|prev_gcd.1.glacial_spike)
+actions.cleave_ss+=/ice_nova,if=freezable&!prev.1.freeze&(prev_gcd.1.glacial_spike)
+actions.cleave_ss+=/frozen_orb,if=cooldown_react
+actions.cleave_ss+=/shifting_power,if=cooldown.icy_veins.remains>10&cooldown.flurry.remains&(boss&fight_remains>cooldown.icy_veins.remains-6)
+actions.cleave_ss+=/cone_of_cold,if=talent.coldest_snap&cooldown.comet_storm.remains>10&cooldown.frozen_orb.remains>10&remaining_winters_chill=0&active_enemies>=3
+actions.cleave_ss+=/glacial_spike,if=buff.icicles.stacks=5&((action.flurry.cooldown_react|remaining_winters_chill)|freezable&(cooldown.freeze.ready|cooldown.ice_nova.ready))
+actions.cleave_ss+=/ray_of_frost,if=remaining_winters_chill&buff.icy_veins.down
+actions.cleave_ss+=/frostbolt,if=talent.deaths_chill&(!talent.freezing_rain&buff.icy_veins.remains>8&buff.deaths_chill.stack<=13|talent.freezing_rain&buff.icy_veins.remains>22)
+actions.cleave_ss+=/ice_lance,if=buff.fingers_of_frost.up&!prev_gcd.1.glacial_spike
+actions.cleave_ss+=/blizzard,if=talent.ice_caller&talent.freezing_rain&active_enemies>=3
+actions.cleave_ss+=/frostbolt
+actions.cleave_ss+=/call_action_list,name=movement
 
 actions.movement+=/any_blink,if=movement.distance>10
 actions.movement+=/ice_floes,if=buff.ice_floes.down
 actions.movement+=/ice_nova
-actions.movement+=/cone_of_cold,if=target.within10&!talent.coldest_snap&active_enemies>=2
+actions.movement+=/cone_of_cold,if=!talent.coldest_snap&active_enemies>=2
 actions.movement+=/arcane_explosion,if=mana.pct>30&active_enemies>=2
 actions.movement+=/fire_blast
 actions.movement+=/ice_lance
 
-actions.ss_st+=/flurry,if=cooldown_react&remaining_winters_chill=0&debuff.winters_chill.down&(prev_gcd.1.frostbolt|prev_gcd.1.glacial_spike)
-actions.ss_st+=/ice_lance,if=buff.icy_veins.up&(debuff.winters_chill.stack=2|debuff.winters_chill.stack=1&debuff.embedded_frost_splinter.stack_pct=100)
-actions.ss_st+=/ray_of_frost,if=buff.icy_veins.down&buff.freezing_winds.down&remaining_winters_chill=1
-actions.ss_st+=/frozen_orb
-actions.ss_st+=/shifting_power
-actions.ss_st+=/ice_lance,if=remaining_winters_chill
-actions.ss_st+=/comet_storm,if=prev_gcd.1.flurry|prev_gcd.1.cone_of_cold|debuff.embedded_frost_splinter.stack_pct=100
-actions.ss_st+=/glacial_spike,if=buff.icicles.stack=5
-actions.ss_st+=/flurry,if=buff.icy_veins.up&debuff.embedded_frost_splinter.stack_pct<100
-actions.ss_st+=/ice_lance,if=buff.fingers_of_frost.react
-actions.ss_st+=/frostbolt
-actions.ss_st+=/call_action_list,name=movement
+actions.st_aoebuild+=/comet_storm,if=prev_gcd.1.flurry&(buff.icy_veins.down|talent.frostfire_bolt)
+actions.st_aoebuild+=/flurry,if=cooldown_react&(buff.icicles.stacks<5|talent.splinterstorm)&(remaining_winters_chill=0&debuff.winters_chill.down&(prev_gcd.1.frostbolt|prev_gcd.1.frostfire_bolt|prev_gcd.1.glacial_spike)|buff.excess_frost.react)
```

Comment:
See previous discussion of `.react`.

Reply:
![alt text](image.png)

In my forthcoming revision